packages/flutter/test/flutter_test_config.dart

```diff
@@ -52,6 +53,8 @@ Future<void> testExecutable(FutureOr<void> Function() testMain) {
     );
   }
 
+  tearDown(() => disposeFlutterSingletons());
```

Comment:
How are the singletons un-disposed for the next test case?

Reply:
Oh, I see they auto-resurrect themselves next time they are accessed. This feels somewhat strange and surprising. I wonder if this could cause hard-to-debug bugs when the identity of a singleton suddenly changes...